### DISCUSSION TITLE
signal error from composer script

### DISCUSSION THREAD

**Discussion by staabm** [composer/composer#10713] [Q&A / Support] [answered]
we have a command class, which we invoke from within a composer-script via composer.json like

```php
/**
 * Command to be used from the composer.json script-events.
 */
final class GruntCommand
{
    /**
     * @return void
     */
    public static function run(\Composer\Script\Event $event)
    {
        $cmd = 'grunt prod';
        if ($event->isDevMode()) {
            $cmd = 'grunt';
        }

        $event->getIO()->writeError($cmd);

        system($cmd);
    }
}
```

how to signal a error to the composer process, so the script event is considered not succesfull and the composer command returns a non-zero exit code?

is it allowed to `exit(1)` or similar from such a script event? 
or is there another way/API, because composer doesn't like beeing interrupted that hard by a script-event?

**Comment by stof** [comment, marked as answer]
you should rather throw an exception.
